### PR TITLE
[ALLUXIO-2211] Fix recursive loadMetadata of a Swift directory

### PR DESCRIPTION
**Pull request by madanadit**
This addresses https://alluxio.atlassian.net/browse/ALLUXIO-2211.

The core problem is that FileSystemMaster:loadMetadataAndJournal calls list as follows:
String[] files = ufs.list(ufsUri.getPath());

getPath eliminates the 'swift://container/' scheme from the uri. The Swift UFS and some others I presume expect that the scheme is present. Although others are more robust to an absence. This PR makes Swift more robust.

I will address correct loadMetadataAndJournal to pass the fully qualified path to a UFS in a later PR which establishes a contract with all UFSs.

### CODE REVIEW

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java [hunk 1]
```diff
@@ -594,6 +594,10 @@ private boolean isDirectory(String path) throws IOException {
       }
     }
 
+    if (self.length() == 0) {
```
Comment (calvinjia) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: Is `isRoot(self)` a more logical condition to check?

Comment (calvinjia) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: Is `isRoot(self)` a more logical condition to check?

**Comment by calvinjia** [comment]
@madanadit Thanks for the fix, is it possible to unit test this?

**Comment by madanadit** [comment]
@calvinjia Thanks for the feedback. I've created a unit test to check if direct children metadata is loaded from UFS on listStatus.

**Comment by peisun1115** [comment]
LGTM. thanks for the fix.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**